hashfs/hashfs.py

```diff
@@ -226,7 +226,10 @@ def haspath(self, path):
 
     def makepath(self, path):
         """Physically create the folder path on disk."""
-        mkpath(path, mode=self.dmode)
+        if not exists(path):
+            os.makedirs(path, self.dmode)
```

Comment:
This introduces a race condition since directory could be created in between `os.path.exists` call and `os.makedirs` call. Better to do something like:

```python
try:
    os.makedirs(path, self.dmode)
except FileExistsError:
    assert os.path.isdir(path), 'expected {} to be a directory'.format(path)
```

If/when Python 2.7 support is dropped, then could use `os.makedirs(path, self.dmode, exist_ok=True)` option.